pymove/utils/integration.py

```diff
@@ -0,0 +1,758 @@
+                
+                # First iteration is minimum distances 
+                if idx == 0:            
+                    minimum_distances[indexs] = ut.distances.haversine(lat_cvp, lon_cvp, df_filted['lat'].to_numpy(dtype=np.float32), df_filted['lon'].to_numpy(dtype=np.float32))
+                    event_id[indexs] = row.event_id
+                    event_type[indexs] = row.event_type
+                else:
+                    current_distances[indexs] = ut.distances.haversine(lat_cvp, lon_cvp, df_filted['lat'].to_numpy(dtype=np.float32), df_filted['lon'].to_numpy(dtype=np.float32))
+                    compare = current_distances < minimum_distances
+                    index_True = np.where(compare==True)[0]
+                    
+                    minimum_distances = np.minimum(current_distances, minimum_distances)  
+                    event_id[index_True] = row.event_id
+                    event_type[index_True] = row.event_type
+                    
+        df_['event_id'] = event_id        
+        df_['dist_cvp'] = minimum_distances
+        df_['event_type'] = event_type 
+        print('Integration with CVP was completed')
+        print('\nTotal Time: {:.2f} seconds'.format((time.time() - start_time)))
+        print('-----------------------------------------------------\n')
+
+    except Exception as e:
+        print('id: {}\n'.format(idx))
+        raise e
+
+def join_with_HOME_by_id(df_, df_home, label_id='id', drop_id_without_home=False):
+    
+    """
+    It performs the integration between trajectories and home points,
+    generating new columns referring to the distance of the nearest
+    home point, address and city of each trajectory point.
+    
+    Parameters
+    ----------
+    df_ : dataframe
+        The input trajectory data.
+    
+    df_home : dataframe
+        The input home points data.
+        
+    label_id : String, optional("id" by default)
+        Label of df_home referring to the home point id.
+        
+    drop_id_without_home : Boolean, optional(False by default)
+        flag as an option to drop id's that don't have houses.
+        
+    Returns
+    -------
+    """
+    
+    try:   
+        print('Integration with Home...')
+        ids_without_home = []
+
+        if df_.index.name is None:
+            print('...setting {} as index'.format(label_id))
+            df_.set_index(label_id, inplace=True)
+
+        for idx in df_.index.unique():
+            filter_home = (df_home[label_id] == idx) 
+            
+            if df_home[filter_home].shape[0] == 0:
+                print('...id: {} has not HOME'.format(idx))
+                ids_without_home.append(idx)
+            else:
+                home = df_home[filter_home].iloc[0]
+                lat_tnz = df_.at[idx, 'lat']
+                lon_tnz = df_.at[idx, 'lon']
+
+                # if tnz has a single tuple
+                if type(lat_tnz) is not np.ndarray:
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Using type() instead of isinstance() for a typecheck.](https://app.codacy.com/gh/InsightLab/PyMove/pullRequest?prid=5610065)